### PR TITLE
Added a feature to update an existing cart

### PR DESCRIPTION
**Pull request by gvieira-dutra**
Added a Put route to the ShoppingCart Features, this allows the API to update the number of Items and their price on an existing cart.

### CODE REVIEW

File: Services/ShoppingCart/ShoppingCart.Api/Data/CachedShoppingCartRepository.cs [hunk 1]
```diff
@@ -37,4 +45,5 @@ public async Task<Cart> GetAsync(string userName, CancellationToken cancellation
 
         return cart;
     }
+
```
Comment (sunnynagavo) [inline Services/ShoppingCart/ShoppingCart.Api/Data/CachedShoppingCartRepository.cs]: nit: remove this extra line

File: Services/ShoppingCart/ShoppingCart.Api/Data/IShoppingCartRepository.cs [hunk 2]
```diff
@@ -3,6 +3,8 @@
 public interface IShoppingCartRepository
 {
     Task<Cart> CreateAsync(Cart cart, CancellationToken cancellationToken = default);
+    Task<Cart> UpdateAsync(Cart cart, CancellationToken cancellationToken = default);
     Task<Cart> GetAsync(string userName, CancellationToken cancellationToken = default);
     Task<bool> DeleteAsync(string userName, CancellationToken cancellationToken = default);
+
```
Comment (sunnynagavo) [inline Services/ShoppingCart/ShoppingCart.Api/Data/IShoppingCartRepository.cs]: +1, remove this extra line

**Review by sunnynagavo — APPROVED** [review]
rest everything Looks good to me

**Comment by gvieira-dutra** [comment]
thanks for the feedback, I removed the extra lines

File: Services/ShoppingCart/ShoppingCart.Api/Features/ShoppingCart/UpdateShoppingCart/UpdateShoppingCartEndpoint.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    {
+        app.MapPut("/api/shopping-cart", async (UpdateShoppingCartRequest request, ISender sender) =>
+        {
+            var command = request.Adapt<UpdateShoppingCartCommand>();
+
+            var result = await sender.Send(command);
+
+            var response = result.Adapt<UpdateShoppingCartResponse>();
+
+            return Results.Created($"/api/shopping-cart/{response.UserName}",
```
Comment (poorna-soysa) [inline Services/ShoppingCart/ShoppingCart.Api/Features/ShoppingCart/UpdateShoppingCart/UpdateShoppingCartEndpoint.cs]: Return should be Results.OK() or Results.NoContent();

File: Services/ShoppingCart/ShoppingCart.Api/Features/ShoppingCart/UpdateShoppingCart/UpdateShoppingCartEndpoint.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+                response);
+        })
+        .WithName("UpdateShoppingCart")
+        .Produces<UpdateShoppingCartResponse>(StatusCodes.Status201Created)
```
Comment (poorna-soysa) [inline Services/ShoppingCart/ShoppingCart.Api/Features/ShoppingCart/UpdateShoppingCart/UpdateShoppingCartEndpoint.cs]: Status either 200 or 204.

**Review by poorna-soysa — CHANGES_REQUESTED** [review]

File: Services/ShoppingCart/ShoppingCart.Api/Features/ShoppingCart/UpdateShoppingCart/UpdateShoppingCartEndpoint.cs [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+public sealed record UpdateShoppingCartRequest(string UserName, List<UpdateShoppingCartItem> Items);
+public sealed record UpdateShoppingCartItem(string ProductCode, int Quantity, decimal Price);
+public sealed record UpdateShoppingCartResponse(string UserName);
+
+public sealed class UpdateShoppingCartEndpoint : ICarterModule
+{
+    public void AddRoutes(IEndpointRouteBuilder app)
+    {
+        app.MapPut("/api/shopping-cart", async (UpdateShoppingCartRequest request, ISender sender) =>
```
Comment (poorna-soysa) [inline Services/ShoppingCart/ShoppingCart.Api/Features/ShoppingCart/UpdateShoppingCart/UpdateShoppingCartEndpoint.cs]: Update endpoint should be /api/shopping-cart/{userName}

File: Services/ShoppingCart/ShoppingCart.Api/Features/ShoppingCart/UpdateShoppingCart/UpdateShoppingCartEndpoint.cs [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+            var result = await sender.Send(command);
+
+            var response = result.Adapt<UpdateShoppingCartResponse>();
+
+            return Results.Ok();
+        })
+        .WithName("UpdateShoppingCart")
+        .Produces<UpdateShoppingCartResponse>(StatusCodes.Status200OK)
+        .ProducesProblem(StatusCodes.Status400BadRequest)
```
Comment (poorna-soysa) [inline Services/ShoppingCart/ShoppingCart.Api/Features/ShoppingCart/UpdateShoppingCart/UpdateShoppingCartEndpoint.cs:23]: SHould be .ProducesProblem(StatusCodes.Status400BadRequest) and .ProducesProblem(StatusCodes.Status404NotFound)

**Review by poorna-soysa — CHANGES_REQUESTED** [review]

**Review by poorna-soysa — APPROVED** [review]

**Pull request merged**